crates/precompile/src/bn128/matter_labs.rs

```diff
@@ -0,0 +1,180 @@
+use super::{PrecompileError, FQ_LEN, G1_LEN, G2_LEN, SCALAR_LEN};
+use eth_pairings::{
+    engines::bn254::*,
+    extension_towers::fp12_as_2_over3_over_2::Fp12,
+    field::U256Repr,
+    integers::MaxGroupSizeUint,
+    pairings::PairingEngine,
+    public_interface::{decode_g1, decode_g2},
+    traits::{Group, ZeroAndOne},
+};
+
+/// G1Point is the concrete representation of a G1 element
+pub(super) type G1Point = eth_pairings::weierstrass::curve::CurvePoint<
+    'static,
+    eth_pairings::weierstrass::CurveOverFpParameters<
+        'static,
+        U256Repr,
+        eth_pairings::field::PrimeField<U256Repr>,
+    >,
+>;
+
+/// G2Point is the concrete representation of a G2 element
+pub(super) type G2Point = eth_pairings::weierstrass::curve::CurvePoint<
+    'static,
+    eth_pairings::weierstrass::CurveOverFp2Parameters<
+        'static,
+        U256Repr,
+        eth_pairings::field::PrimeField<U256Repr>,
+    >,
+>;
```

Comment:
Types in matter-labs-eip1962 are complicated and not re-exported like substrate-bn